### PR TITLE
Implement Method To Delete School

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Notion ticket link

[Delete school](https://www.notion.so/uwblueprintexecs/Delete-school-5740095fef174afe837c49ec9bf71e39)

## Implementation description
* Implemented a method to delete a school by its ID
* Added unit tests for new method

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/implementations/__tests__/schoolService.test.ts [hunk 1]
```diff
@@ -167,4 +167,20 @@ describe("mongo schoolService", (): void => {
       `School id ${notFoundId} not found`,
     );
   });
+
+  it("deleteSchool", async () => {
+    const savedSchool = await SchoolModel.create(testSchools[0]);
+
+    const deletedSchool = await schoolService.deleteSchool(savedSchool.id);
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/schoolService.test.ts]:
```suggestion
    const deletedSchoolId = await schoolService.deleteSchool(savedSchool.id);
```
nit: variable rename

File: backend/services/implementations/__tests__/schoolService.test.ts [hunk 2]
```diff
@@ -167,4 +167,20 @@ describe("mongo schoolService", (): void => {
...
+    expect(deletedSchool).toBe(savedSchool.id);
+  });
+
+  it("deleteSchool with non-existing id", async () => {
+    await expect(async () => {
+      const notFoundId = "86cb91bdc3464f14678934cd";
+      await schoolService.deleteSchool(notFoundId);
+    }).rejects.toThrowError(
+      `School with id 86cb91bdc3464f14678934cd not found`,
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/schoolService.test.ts]:
```suggestion
      `School with id ${notFoundId} not found`,
```

File: backend/services/implementations/schoolService.ts [hunk 3]
```diff
@@ -172,6 +173,23 @@ class SchoolService implements ISchoolService {
       throw error;
     }
   }
+
+  /* eslint-disable class-methods-use-this */
+  async deleteSchool(id: string): Promise<string> {
+    try {
+      const deletedSchool = await MgSchool.findByIdAndDelete(id);
```
Comment (carissa-tang) [inline backend/services/implementations/schoolService.ts]:
```suggestion
      const deletedSchool: School | null = await MgSchool.findByIdAndDelete(id);
```

**Review by carissa-tang — APPROVED** [review]
added some nit comments.. but rest looks good!

**Review by joyce-shi — APPROVED** [review]

**Pull request merged**